packages/rpc/src/adaptors/message-port.ts

```diff
@@ -1,6 +1,7 @@
 import type {MessageEndpoint} from '../types';
 
 export function fromMessagePort(messagePort: MessagePort): MessageEndpoint {
+  messagePort.start();
```

Comment:
I don't think so, `.start()` just needs to be called before messages will be sent properly.  It doesn't matter if you add the change listener before or after calling `.start()`

Reply:
I am still concerned about this, to be honest. When we receive the message port, we don't know if the other side has started to send important messages or not. If we call `start()` and there are already important messages, we will miss them, because code after this line adds the `message` event listener.

I tried to simulate this using message ports in the browser and it confirmed the behavior of "losing" events when calling `start()` too early:

<img width="634" alt="Screen Shot 2022-02-25 at 2 43 48 PM" src="https://user-images.githubusercontent.com/3012583/155786122-9d658562-7c85-4672-8250-9ab0ff564ad7.png">
<img width="338" alt="Screen Shot 2022-02-25 at 2 44 41 PM" src="https://user-images.githubusercontent.com/3012583/155786171-d4ee04b5-3e25-4a0a-847e-f27bdeb882b5.png">

In both cases, a message is sent from `port2` before the listener is added on `port1`. In the first screenshot, we can still get access to the message because the event listener is added before `start()` is called. In the second screenshot, the message is lost entirely.

Reply:
Oh I see, that makes sense.  Thanks for explaining the issue that way.  I'll think about it and see if I can find a way to address this, and add some more unit tests covering this situation if I can.